### PR TITLE
firstaid for arydshln

### PR DESCRIPTION
**Pull request by davidcarlisle**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [X] Test file(s) added
- [X] Version and date string updated in changed source files
- [X] Relevant `\changes` entries in source included
- [X] Relevant `changes.txt` updated
- [ ] Rollback provided (if necessary)?
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: required/firstaid/latex2e-first-aid-for-external-files.dtx [hunk 1]
```diff
@@ -721,6 +721,30 @@
 }
 %    \end{macrocode}
 %
+% \subsection{The \pkg{arydshln} package first aid}
+%
+%    The package does not declare a \texttt{prop}, which causes an issue with newer
+%    routines in \pkg{expl3}.
```
[the remark this answers is not in the document]
Comment (davidcarlisle) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:727]: I didn't write that let me check the diff (probably accidental from an aborted diff from develop)

File: required/firstaid/changes.txt [hunk 2]
```diff
@@ -1,3 +1,8 @@
+2024-06-25  David Carlisle  <David.Carlisle@latex-project.org>
+
+	* latex2e-first-aid-for-external-files.dtx: add firstaid for arydshln
+	\protect macros to be safe in colortbl
```
Comment (muzimuzhi) [inline required/firstaid/changes.txt:4]: This `\protect` could be misleading, as the actual firstaid uses `\protected`. Use "Protect macros ..." or "Make macros \protected ..."?

**Comment by muzimuzhi** [comment]
BTW, CI job named "[Tests: required/firstaid](https://github.com/latex3/latex2e/actions/runs/9668636085/job/26673319670#logs)" failed with ``! LaTeX Error: File `arydshln.sty' not found.``.

Package `arydshln` needs to be added to `.github/tl_packages`.
https://github.com/latex3/latex2e/blob/d0e2e6eceb176ca8bf40c67128bcec73255465fb/.github/tl_packages#L28C1-L28C31

**Comment by davidcarlisle** [comment]
> BTW, CI job named "[Tests: required/firstaid](https://github.com/latex3/latex2e/actions/runs/9668636085/job/26673319670#logs)" failed with `` ! LaTeX Error: File `arydshln.sty' not found. ``.
> 
> Package `arydshln` needs to be added to `.github/tl_packages`.
> 
> https://github.com/latex3/latex2e/blob/d0e2e6eceb176ca8bf40c67128bcec73255465fb/.github/tl_packages#L28-L

 I just saw that failed, thanks for providing the instant  fix

**Review by FrankMittelbach — APPROVED** [review]

**Pull request merged**